File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int answer=0,profit=0,temp=INT_MAX;
        for(int i=0;i<prices.size();i++)
        {
            if(prices[i]<temp)
                temp=prices[i];
            profit=prices[i]-temp;
            if(profit>answer)
                answer=profit;
    }
        return answer;
        
        
    }
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left,right=0,1
        maxprof=0
        while  right < len(prices):
            if prices[left] < prices[right]:
                prof=prices[right]-prices[left]
                maxprof=max(prof,maxprof)
            else:
                left=right
            right+=1
        return maxprof